ldraw_to_scad.py

```diff
@@ -101,14 +98,54 @@ def process_main(self, input_lines):
                 completed.append(current_module.get_module_name())
         # Now we can create output lines - starting at the top
         # of completed modules.
-        output_lines = []
+        output_lines = ["""
+module makepoly(poly)
+    for(p=poly)
+        color(lego_colours[p[2]])
+            polyhedron(p[0], p[1]);
+
+function l1(M, polys, col, invert=false) =
+    [for(poly=polys) [
+        [for(p=poly[0]) v3(M * [p.x, p.y, p.z, 1])],
+        [for(face=poly[1]) ((
```

Comment:
Might be good to have a comment around this - having a hard time understanding it but it looks like it's pulling scalars out of a matrix and calculating the dot product of one minus the dot product of the other?